### PR TITLE
Update "Coming Episodes" to the same splitting on the web and in the API

### PR DESCRIPTION
**Pull request by MGaetan89**
I'm not 100% sure about this yet, so don't hesitate to comment if you see something wrong.

@fernandog

### CODE REVIEW

**Comment by fernandog** [comment]
why not sure?

**Comment by MGaetan89** [comment]
Why I tested it in a "non-calendar" layout, I had one "Missed" section with one episode and one "Sunday" section with 3 missed episodes in it (the background color was red as expected). I don't know if it was like that before or not.

Secondly, on the web UI, `today` is defined as `datetime.datetime.now().replace(tzinfo=network_timezones.sb_timezone)`, while it is simply `date.today().toordinal()` in the API. Might need to have a look at that too.

**Comment by fernandog** [comment]
about today, if airs today and airdate < now, keep as today, otherwise missed. isn't?

File: sickrage/show/ComingEpisodes.py [hunk 1]
```diff
@@ -158,3 +158,21 @@ def get_coming_episodes(categories, sort, group, paused=sickbeard.COMING_EPS_DIS
...
+        runtime = data['runtime']
+
+        if runtime:
+            air_date += timedelta(minutes=runtime)
+
+        return air_date.toordinal() < date.today().toordinal()
+
+    @staticmethod
+    def is_episode_today(data):
+        return data['localtime'].toordinal() == date.today().toordinal()
```
Comment (miigotu) [inline sickrage/show/ComingEpisodes.py]: This last function is the only one I see that looks wrong at first glance xD.

is_episode_today is checking if the time is at exactly a day it looks like, no hours, no seconds.

I think you want something like:

```python
def is_episode_today(data):
        return (data['localtime'] - date.today()).days == 0
```

**Comment by MGaetan89** [comment]
Thanks for the feedbacks.
I update the `is_episode_today()` method according to @miigotu suggestion.

**Comment by fernandog** [comment]
Should we merge?

**Comment by MGaetan89** [comment]
If you don't see anything wrong, you can.

**Comment by miigotu** [comment]
let's wait, need to review this with rested eyes later

**Comment by fernandog** [comment]
@MGaetan89 

```
Error !

TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'datetime.date'

<tbody style="text-shadow:none;"> % for cur_result in results: <% if int(cur_result['paused']) and not sickbeard.COMING_EPS_DISPLAY_PAUSED: continue cur_indexer = int(cur_result['indexer'])
/home/osmc/SickRage/sickrage/show/ComingEpisodes.py, line 178:
return (data['localtime'] - date.today()).days == 0
/home/osmc/SickRage/gui/slick/views/comingEpisodes.mako, line 97:
<%
/home/osmc/SickRage/gui/slick/views/layouts/main.mako, line 220:
/home/osmc/SickRage/lib/mako/runtime.py, line 853:
callable_(context, *args, **kwargs)
```

**Comment by miigotu** [comment]
should be `- datetime.now()` or `- datetime.today()` then.

**Comment by MGaetan89** [comment]
I will look again at this after I'm done on the API part. Shouldn't be too long now.

**Comment by fernandog** [comment]
@MGaetan89 shouldn't be ?

return (data['localtime'] - datetime.date.today()).days == 0

**Comment by MGaetan89** [comment]
No, I don't think so. Because I do `from datetime import date` and not `import datetime`.

**Comment by fernandog** [comment]
@MGaetan89 
let me know when you need to test this. sunday/monday are great days to test

**Comment by MGaetan89** [comment]
I will go through all the comment related to this PR tomorrow to update it accordingly. But I probably won't be done before the end of the day. Maybe it will be by Tuesday evening.

**Comment by fernandog** [comment]
NP!

**Comment by fernandog** [comment]
@MGaetan89 thinking about this. We should not check ep status in comingEpisodes.

It user dailysearcher freq is 3 hours, comingepisodes wont  sjow as wanted. CE is like a calendar. Not ep status managment. Agree?

**Comment by fernandog** [comment]
all datetimes should use network_timezones.sb_timezone, isn't?

**Comment by miigotu** [comment]
No, it has to check episode status, because skipped and ignored are not included in the results. And no, network_timezones.sb_timezone has nothing to do with anything other than refreshing from indexer.

The object is to clean up the bad code not just add it right back after we refactor it.

**Comment by fernandog** [comment]
I only want my airdates with my local time. Thats why
We should convert unaired>wanted every full hour then.
Some users like @neoatomic sets dailysearch to 240min. ComingEpisodes will be wrong. 4 hours delayed. 
ComingEpisodes IMO is a calender. Episode status user could check in displayShow.

**Comment by miigotu** [comment]
"I only want my airdates with my local time." no shit, so why would you need to convert anything when the airdates are already in a proper format.

Coming episodes are only for shows you WANT or are UNAIRED. It has always been this way.

**Comment by fernandog** [comment]
Did I asked something diff  ?!
Just said to dont get ep status (unaired/wanted). But is beter to use airdates instead of ep status (as I said before)
Who said something about ignored?!

**Comment by fernandog** [comment]
just to explain in details what I previous said (as It's not a good ideia as it depends on dailysearch freq):
get all WANTED ep status and show in comingEpisodes.
all the UNAIRED would be classified: today, soon, later

will start to post full details of my comments as they are get misinterpreted

**Comment by miigotu** [comment]
wanted is not only coming. they can be missed, soon, or current. everybody else understands how it is supposed to be.

**Comment by fernandog** [comment]
AGAIN:
get all WANTED ep status and show in comingEpisodes.
all the UNAIRED would be classified: today, soon, later

**Comment by miigotu** [comment]
that's how it always has and will work. so why comment to argue something when its already how it is. just don't say anything then.

**Comment by miigotu** [comment]
like how you originally said to not check status, now you say to get the status.

**Comment by fernandog** [comment]
(as It's not a good ideia as it depends on dailysearch freq):

please stop, people are already complaining about how you reply uses

**Comment by fernandog** [comment]
@MGaetan89 everything ok? We didnt hear from you anymore

**Comment by MGaetan89** [comment]
@fernandog Yes everything's ok. Just gonna be very busy this month. So I won't have a lot of time for SickRage until November. I focus my free time on my app until then.

**Comment by miigotu** [comment]
Can this one be updated or did you change this in another PR or plan to....?

We have lots of PR open atm that would be nice to get completed/merged.

**Comment by fernandog** [comment]
@MGaetan89 when you do plan to finish this PR? thanks!

**Comment by MGaetan89** [comment]
As I said a couple of weeks back, I don't have a lot of free time this month.
So I will help working on SickRage next month (so either next week or in two weeks).

**Comment by MGaetan89** [comment]
I rebased this with the latest version of `develop`. Hope I didn't lost anything in the process.
I will try to (finally) finish this.

**Pull request closed without merging**